modules/collect/src/main/java/com/opengamma/strata/collect/Guavate.java

```diff
@@ -69,6 +69,30 @@ private Guavate() {
 
   //-------------------------------------------------------------------------
   /**
+   * Uses a number of suppliers to create a single optional result.
+   * <p>
+   * The Java 8 {@link Optional} class does not have an {@code or} method,
+   * so this provides an alternative.
+   * 
+   * @param <T>  the type of element in the iterable
+   * @param suppliers  the suppliers to combine
+   * @return the list that combines the inputs
+   */
+  @SuppressWarnings("unchecked")
+  @SafeVarargs
+  public static <T> Optional<T> firstNotEmpty(Supplier<Optional<? extends T>>... suppliers) {
```

Comment:
This method is useful for the use cases in this PR, but would it be worth adding an overload with the signature:

    Optional<T> firstNotEmpty(Optional<? extends T>... optionals)

That might be more generally useful.